### PR TITLE
Working plotly output implementation

### PR DESCRIPTION
**Pull request by kevcunnane**
- Essentially has min width of 700px with horizontal scroll if smaller
- Renders outputs as expected

image
image

Example of when width is < 700px:
image

### CODE REVIEW

**Comment by anthonydresser** [comment]
Whats the long term goal here? For every new output type users want we will have to hard code in another render type? increasing our package size and decreasing our start up performance? This doesn't seem like a maintainable approach.

File: package.json [hunk 1]
```diff
@@ -60,6 +61,7 @@
     "native-watchdog": "1.0.0",
     "ng2-charts": "^1.6.0",
     "node-pty": "0.9.0-beta9",
+    "plotly.js-dist": "^1.48.3",
```
Comment (kburtram) [inline package.json]: Did you see any startup time impact from adding this dependency?  We need to be somewhat careful if this gets loaded on the startup path.

File: src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		//   return cloneDeep(figure) as Figure;
+		// }
+
+		const { data = [], layout = {} } = figure;
+
+		return { data, layout };
+	}
+
+	private displayError(error: Error | string): void {
+		this.errorText = localize('plotlyError', 'Error displaying Plotly graph: {0}', getErrorMessage(error));
```
Comment (chlafreniere) [inline src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts]: double quotes 😄

File: src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	}
+
+	layout(): void {
+		// if (this.mimeType === 'text/vnd.plotly.v1+html') {
+		// 	// Do nothing - this is our way to ignore the offline init Plotly attempts to do via a <script> tag.
```
Comment (chlafreniere) [inline src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts]: Looks like there are some double comments here

File: package.json (hunk 1 above)
Comment (kevcunnane) [reply package.json]: I'm moving it to dynamic load inside the component using require (after chatting to Anthony). Verified this only loads in the module on open of a notebook with a plotly component. Therefore now has no startup impact. thanks for this feedback!

File: src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts (hunk 2 above)
Comment (kevcunnane) [reply src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts]: Do I owe you a drink for every one of these I miss?

File: src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts (hunk 3 above)
Comment (kevcunnane) [reply src/sql/workbench/parts/notebook/outputs/plotlyOutput.component.ts]: Yeah removing all this - fixed

**Comment by kevcunnane** [comment]
@anthonydresser I believe I've addressed your ask (not impact startup perf), plus for package size I picked the "dist" option which is 4 files and the minimum size. Given support for Plotly is a high request from PM this seems like a reasonable solution.

Longer term, we will be adding in new renderers and the top ones should be in-app. We can look at webviews for contributing lesser-known ones, but I don't know of a good tradeoff since more than a couple webviews have a big perf hit (e.g. 30mb per view)

**Pull request merged**